library/ssl_tls13_generic.c

```diff
@@ -1484,9 +1484,12 @@ int mbedtls_ssl_tls13_check_early_data_len(mbedtls_ssl_context *ssl,
          ssl->total_early_data_size)) {
 
         MBEDTLS_SSL_DEBUG_MSG(
-            2, ("EarlyData: Too much early data received, %u + %" MBEDTLS_PRINTF_SIZET " > %u",
-                ssl->total_early_data_size, early_data_len,
-                ssl->session_negotiate->max_early_data_size));
+            2, ("EarlyData: Too much early data received, "
+                "%" MBEDTLS_PRINTF_SIZET " + %" MBEDTLS_PRINTF_SIZET
+                " > %" MBEDTLS_PRINTF_SIZET,
+                (size_t) ssl->total_early_data_size,
```

Comment:
If we're going to cast a `uint32_t`, why not cast to `long` so we can just use `%lu` - smaller change and the resulting code is easier to read (I mean, given we have an existing `MBEDTLS_PRINTF_SIZET` it's not great at the moment, of course)

Reply:
I don't think mixing types makes the instruction as a whole more readable, but it works either way.